homeassistant/components/geniushub/__init__.py

```diff
@@ -54,20 +69,37 @@
 
 MAC_ADDRESS_REGEXP = r"^([0-9A-F]{2}:){5}([0-9A-F]{2})$"
 
+data = {
+    CONF_MAC: DEFAULT_GENIISHUB_MAC,
+    CONF_TOKEN: DEFAULT_GENIISHUB_TOKEN,
+    CONF_HOST: DEFAULT_GENIUSHUB_HOST,
+    CONF_PASSWORD: DEFAULT_GENIUSHUB_PASSWORD,
+    CONF_USERNAME: DEFAULT_GENIUSHUB_USERNAME,
+}
+
 V1_API_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_TOKEN): cv.string,
-        vol.Required(CONF_MAC): vol.Match(MAC_ADDRESS_REGEXP),
-    }
+        vol.Required(CONF_TOKEN, default=data.get(CONF_TOKEN)): cv.string,
+        vol.Required(
+            CONF_MAC, default=data.get(CONF_MAC)
+        ): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
+    },
+    extra=vol.PREVENT_EXTRA,
 )
+
+
 V3_API_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_MAC): vol.Match(MAC_ADDRESS_REGEXP),
-    }
+        vol.Required(CONF_HOST, default=data.get(CONF_HOST)): cv.string,
+        vol.Required(CONF_USERNAME, default=data.get(CONF_USERNAME)): cv.string,
+        vol.Required(CONF_PASSWORD, default=data.get(CONF_PASSWORD)): cv.string,
+        vol.Optional(
+            CONF_MAC, default=data.get(CONF_MAC)
+        ): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
+    },
+    extra=vol.PREVENT_EXTRA,
 )
```

Comment:
Can we keep this like the original?

Reply:
Almost: ```vol.Optional(CONF_MAC): vol.Match(MAC_ADDRESS_REGEXP),``` throws an error when the UI is displayed.
```vol.Optional(CONF_MAC): cv.string``` works.